avocado/core/settings.py

```diff
@@ -244,6 +244,11 @@ def get_value(self, section, key, key_type=str, default=no_default,
         if not val.strip() and not allow_blank:
             return self._handle_no_value(section, key, default)
 
+        # if the value looks like a user directory, let's expand it
+        if key_type == str:
+            if isinstance(val, str):
+                val = os.path.expanduser(val)
```

Comment:
Well I don't think settings should magically replace all strings as it might corrupt user values. We could implement `"__PATH__"` key_type which would be `str` with this tweak, though.